pkg/metrics/pushclient/pushclient.go

```diff
@@ -0,0 +1,78 @@
+package pushclient
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+// aggregationClient is a custom HTTP client for use with
+// WeaveWorks Prometheus Aggregation PushGateway
+// We will embed http.Client and essentially
+// "override" the Do function
+type aggregationClient struct {
+	// embed a stdlib http.Client
+	*http.Client
+}
+
+// gatewayURL is the default url that will be used as the endpoint to push to.
+// To push to a centralized prometheus aggregation gateway, change this url.
+var gatewayURL = "http://localhost:9092"
+
+// Do is a custom written client to send data to the prometheus aggregate
+// gateway. It creates a new client and pushes the body that was passed to this function.
+// https://godoc.org/github.com/prometheus/client_golang/prometheus/push#HTTPDoer
+func (a *aggregationClient) Do(req *http.Request) (resp *http.Response, err error) {
```

Comment:
We don't need this aggregation Client anymore as of weaveworks/prom-aggregation-gateway#31, right? 

We think support for a custom Client will be a production requirement for authenticating to telemeter, so we should probably keep the ability to inject a custom client, but just inject the default (without overriding `Do`)

Reply:
Shouldn't we create a custom round tripper inside the Do function? For now we don't need it so I'll default the Do function to call the http.Client Do function.

Reply:
> Shouldn't we create a custom round tripper inside the Do function

@squat will providing a custom implementation of the `Do` function be sufficient to allow authentication to telemeter? when we met with the observability group, it was said we need a "custom roundtripper" but I think this Do function was what was actually meant, because that is all the standard prometheus push client uses...